app/models/name/merge.rb

```diff
@@ -106,6 +112,9 @@ def merge(old_name)
       self.citation = old_name.citation.strip_squeeze
     end
 
+    # Update the identifier if user provided one
+    self.icn_id = old_name.icn_id if old_name.icn_id
```

Comment:
I seem to remember the general logic here is that where a data field is provided by both names, it prefers self over old_name.  That's because usually self has more references / uses in the database, and that old_name is often a one-off typo or mistake.  Self should definitely be the more reliable source.

Therefore I recommend:

    self.icn_id = old_name.icn_id if self.icn_id.blank?